### DISCUSSION TITLE
SvelteKit: SSR object appears briefly, then displays undefined

### DISCUSSION THREAD

**Discussion by renefournier** [supabase/supabase#3610] [Questions]
There is probably a very obvious reason for this error, but I can't see it. In a nutshell, I'm server-side fetching a profile record via Supabase, then _attempting_ to pass it in as props to my `Profile.svelte` page. Doing a `console.dir(profile)` in `<script context="module">` shows the object data, but doing the same in `onMount` displays undefined (even though the data does briefly flash before supposedly Svelte hydration happens and the text turns undefined. Here's a stripped down version of the component:

```
<script context="module">
	import supabase from '$lib/db.js';

	export async function load({ page, fetch, session, context }) {
		if (!session) {
			return {
				status: 302,
				redirect: '/login'
			};
		} else {
			const user = supabase.auth.user();
			const res = await supabase.from('profiles').select('id, name, owner, address');

			let profile = res.data[0];
			console.dir(profile);

			return {
				props: {
					user,
					session,
					profile
				}
			};
		}
	}
</script>

<script>
	import { onMount } from 'svelte';

	export let user;
	export let session;
	export let profile;

	onMount(async () => {
		console.log('🐎');
		console.dir(profile);
	});
</script>

<div class="bg">
	{JSON.stringify(profile)}
</div>
```

Any ideas are much appreciated.

**Comment by renefournier** [comment]
FWIW, I'm using https://github.com/sveltemaster/supasveltekit-cookies-demo/tree/main/src/routes as a guide to **auth** in SvelteKit via Supabase.

**Comment by zakaria-chahboun** [comment]
Don't use `const user = supabase.auth.user();` in the `load` function, Because the load function runs in the server side in the first time, And the  `supabase.auth.user()` grabs the user from the browser in the `localstorage`.

That's why it will give you always `undefined` result 😒!

Then the `load` function runs secondly on the client side!

So! You have to create two ways of authentication, 1 for the SSR and 2 for the Client side, By reading this article 👇
https://www.ashleyconnor.co.uk/2021/06/24/authenticated-server-rendered-pages-with-sveltekit-and-supabase

> Then you can get your user by `load({session})` instead!

After that if you want to login in with a third party provider (like google) and logout (in general) try my solutions 😊:  
https://github.com/supabase/supabase/discussions/2842#discussioncomment-1486453
https://github.com/supabase/supabase/discussions/3468#discussioncomment-1490404

**Comment by michalstruck** [comment]
@zakaria-chahboun This was very obvious, and yet I couldn't see it. Thank you!
